lib/cocoapods/project.rb

```diff
@@ -306,18 +307,49 @@ def group_for_path_in_group(absolute_pathname, group, reflect_file_system_struct
         end
       end
 
-      # Turn .lproj into a variant group
+      # Turn files inside .lproj directories into a variant group
       if relative_dir.basename.to_s =~ lproj_regex
-        filename = absolute_pathname.basename.sub_ext('').to_s
+        group_name = variant_group_name(absolute_pathname)
         lproj_parent_dir = absolute_pathname.dirname.dirname
-        group = @variant_groups_by_path_and_name[[lproj_parent_dir, filename]] ||
-          group.new_variant_group(filename, lproj_parent_dir)
-        @variant_groups_by_path_and_name[[lproj_parent_dir, filename]] ||= group
+        group = @variant_groups_by_path_and_name[[lproj_parent_dir, group_name]] ||
+          group.new_variant_group(group_name, lproj_parent_dir)
+        @variant_groups_by_path_and_name[[lproj_parent_dir, group_name]] ||= group
       end
 
       group
     end
 
+    # Returns the name to be used for a the variant group for a file at a given path.
+    # The path must be localized (within an *.lproj directory).
+    #
+    # @param  [Pathname] The localized path to get a variant group name for.
+    #
+    # @return [String] The variant group name.
+    #
+    def variant_group_name(path)
+      unless path.to_s =~ %r{\.lproj\/}i
+        raise ArgumentError, 'Only localized resources can be added to variant groups.'
+      end
+
+      # When using Base Internationalization for XIBs and Storyboards a strings
+      # file is generated with the same name as the XIB/Storyboard in each .lproj
+      # directory:
+      #   Base.lproj/MyViewController.xib
+      #   fr.lproj/MyViewController.strings
+      #
+      # In this scenario we want the variant group to be the same as the XIB or Storyboard.
+      #
+      # Base Internationalization: https://developer.apple.com/library/ios/documentation/MacOSX/Conceptual/BPInternational/InternationalizingYourUserInterface/InternationalizingYourUserInterface.html
+      if File.extname(path).downcase == '.strings'
+        ['.xib', '.storyboard'].each do |extension|
+          possible_interface_file = Pathname(File.join(path.dirname.dirname, 'Base.lproj', path.basename.sub_ext(extension)))
+          return possible_interface_file.basename.to_s if File.exist?(possible_interface_file)
```

Comment:
FYI besides mocking or real FS fixtures, there’s also faking the FS, e.g. https://github.com/fastlane/deliver/blob/master/spec/loader_spec.rb